src/metabase/driver/bigquery.clj

```diff
@@ -343,14 +347,11 @@
 
 (defn- execute-query [{{{:keys [dataset-id]} :details, :as database} :database, {sql :query, :keys [table-name mbql?]} :native}]
   (let [results (process-native* database sql)
-        results (if mbql?
-                  (post-process-mbql dataset-id table-name results)
-                  results)
-        columns (vec (keys (first results)))]
-    {:columns   columns
-     :rows      (for [row results]
-                  (mapv row columns))
-     :annotate? true}))
+        results (u/prog1 (if mbql?
+                           (post-process-mbql dataset-id table-name results)
+                           (update results :columns (partial map keyword)))
+                  (qp/validate-results <>))]
```

Comment:
is the intention to validate the results of every query execution?  that seems overkill.  and for testing purposes we already have the `post-check-results-format` QP middleware function right?